session1/challenge/client/src/js/components/container/App.js

```diff
@@ -133,12 +143,21 @@ class App extends Component {
   componentDidMount() {
     const socket = socketIOClient(ENDPOINT);
 
+    socket.on("FromAPI", ({data}) => {
+      this.setState((prevState, props) =>  {
+        const {temperature} = data;
+        // Si no existe el data en el estado o la temperatura actual es diferente a la recivida
+        if (prevState.data === null || temperature !== prevState.data.temperature) {
+          console.log(temperature, prevState.data);
+          return {date: new Date()};
+        }
 
-    // TODO: Update the state.date only if the date.temperature is different
-    // Use functional state
+        return null;
+      });
 
-    socket.on("FromAPI", (data) => {
-      this.setState(() => null);
+      this.setState(() =>  {
```

Comment:
Why do you have another call to `setState` here?

Reply:
I wrote that code because i seen the next TODO : 
```js
    // TODO: Update the state.date only if the date.temperature is different
    // Use functional state

    socket.on("FromAPI", (data) => {
      this.setState(() => null);
    });
```
But if I think about it better, I think it was a typing error ("**date**" really is "**data**").

I'm going to change it, am sorry